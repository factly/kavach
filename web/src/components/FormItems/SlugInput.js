import React from 'react';
import { checker } from '../../utils/sluger';
import { Form, Input } from 'antd';

const SlugInput = ({ onChange, inputProps, formItemProps }) => {
  inputProps = onChange ? { ...inputProps, onChange } : { ...inputProps };
  return (
    <Form.Item
      name="slug"
      label="Slug"
      rules={[
        {
          required: true,
          message: 'Please input the slug!',
        },
        {
          pattern: checker,
          message: 'Please enter valid slug!',
        },
      ]}
      {...formItemProps}
    >
      <Input {...inputProps} />
    </Form.Item>
  );
};

export default SlugInput;
