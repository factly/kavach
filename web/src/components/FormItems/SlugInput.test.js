import React from "react";
import { render, shallow } from "enzyme";
import SlugInput from "./SlugInput";

describe('Slug Input Component', () => {
	it('should render and match snapshot', () => {
		const wrapper = render(<SlugInput />);
		expect(wrapper).toBeTruthy();
		expect(wrapper).toMatchSnapshot();
	});

	it('should render with props and match snapshot', () => {
		const props = { onChange: jest.fn(), inputProps: { placeholder: 'test' }, formItemProps: { label: 'test' } };
		const wrapper = render(<SlugInput {...props} />);
		expect(wrapper).toBeTruthy();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render when onChange is not passed', () => {
		const props = { inputProps: { placeholder: 'test' }, formItemProps: { label: 'test' } };
		const wrapper = render(<SlugInput {...props} />);
		expect(wrapper).toBeTruthy();
		expect(wrapper).toMatchSnapshot();
	});
});
