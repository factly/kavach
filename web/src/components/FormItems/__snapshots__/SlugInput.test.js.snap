// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slug Input Component should render and match snapshot 1`] = `
<div
  class="ant-form-item"
>
  <div
    class="ant-row ant-form-item-row"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        for="slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            aria-required="true"
            class="ant-input"
            id="slug"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slug Input Component should render when onChange is not passed 1`] = `
<div
  class="ant-form-item"
>
  <div
    class="ant-row ant-form-item-row"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        for="slug"
        title="test"
      >
        test
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            aria-required="true"
            class="ant-input"
            id="slug"
            placeholder="test"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slug Input Component should render with props and match snapshot 1`] = `
<div
  class="ant-form-item"
>
  <div
    class="ant-row ant-form-item-row"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        for="slug"
        title="test"
      >
        test
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            aria-required="true"
            class="ant-input"
            id="slug"
            placeholder="test"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
